src/persist/EpochAdvancers.cpp

```diff
@@ -68,19 +69,35 @@ DedicatedEpochAdvancer::DedicatedEpochAdvancer(GlobalTestConfig* gtc, EpochSys*
 
 void DedicatedEpochAdvancer::advancer(int task_num){
     EpochSys::init_thread(task_num);// set tid to be the last
+    uint64_t curr_epoch = INIT_EPOCH;
     while(!started.load()){}
     while(started.load()){
-        esys->advance_epoch_dedicated();
-        std::this_thread::sleep_for(std::chrono::microseconds(epoch_length));
+        // wait for sync_signal to fire or timeout
+        std::unique_lock<std::mutex> lk(sync_signal.bell);
+        sync_signal.advancer_ring.wait_for(lk, std::chrono::microseconds(epoch_length), 
```

Comment:
Why do we need a `pred` like this lambda here? Given that `target_epoch` is always incremented by 1 after epoch advancing, this predicate will always be true, right?

Reply:
Oops nice catch. I meant to use that predicate to catch spurious wakeups. I'll apply a little fix.

Reply:
Nice solution. I think now it works. Thanks!